app/controllers/journeys/additional_payments_for_teaching/reminders_form_callbacks.rb

```diff
@@ -0,0 +1,95 @@
+module Journeys
+  module AdditionalPaymentsForTeaching
+    module RemindersFormCallbacks
+      def personal_details_before_show
+        try_mailer { set_a_reminder_immediately_if_possible }
+      end
+
+      def email_verification_before_show
+        try_mailer { set_a_reminder_immediately_if_possible }
+      end
+
+      def email_verification_before_update
+        inject_sent_one_time_password_at_into_the_form
+      end
+
+      def personal_details_after_form_save_success
+        update_reminder_id
+        try_mailer { send_verification_email } || return
+        redirect_to_next_slug
+      end
+
+      def email_verification_after_form_save_success
+        try_mailer { send_reminder_set_email } || return
+        redirect_to_next_slug
+      end
+
+      def personal_details_after_form_save_failure
+        render_template_for_current_slug
+      end
+
+      def email_verification_after_form_save_failure
+        render_template_for_current_slug
+      end
+
+      private
+
+      def set_a_reminder_immediately_if_possible
+        return if current_reminder.persisted?
+
+        if current_reminder.email_verified? && current_reminder.save
+          ReminderMailer.reminder_set(current_reminder).deliver_now
+
+          redirect_to reminder_path(current_journey_routing_name, "set")
+        end
+      end
+
+      def inject_sent_one_time_password_at_into_the_form
+        params[:form]&.[]=(:sent_one_time_password_at, session[:sent_one_time_password_at])
```

Comment:
Could we store the `sent_one_time_password_at` on the `Reminder`? Then we can access it in the form object which might make things a bit simpler

Reply:
I understand your question, and I remember seeing you explain why that was a good idea for the `Claim` model.

My thoughts are that storing `sent_one_time_password_at`,  a transient attribute, directly on the `Reminder` model might introduce unnecessary coupling between the model and the form object. While with the existing approach there is coupling between the controller and the form object, this coupling is more acceptable and expected within the context of MVC architecture. Controllers (and the "state", e.g. in the session) are meant to interact with form objects to handle form submissions and orchestrate the flow of data within the application. Either way I don't feel too strong, and it might make sense to have a consistent approach, but I wouldn't consider it a blocker. Will have a think

Reply:
I'm not sure `sent_one_time_password_at` is transient 🤔 but I don't think it's too big a deal to keep it in the session